src/test/java/StringClassStudyTest.java

```diff
@@ -0,0 +1,69 @@
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class StringClassStudyTest {
+
+    @Test
+    @DisplayName("\"1,2\"을 ,로 split 했을 때 1과 2로 잘 분리되는지 확인하는 학습 테스트를 구현한다.")
+    public void 요구사항1() throws Exception {
+        // given
+        String test = "1,2";
+        // when
+        String[] split = test.split(",");
+
+        // then
+        assertThat(split).containsExactly("1","2");
+    }
+
+    @Test
+    @DisplayName("\"1\"을 ,로 split 했을 때 1만을 포함하는 배열이 반환되는지에 대한 학습 테스트를 구현한다.")
+    public void 요구사항1_2() throws Exception {
+        // given
+        String test = "1";
+
+        // when
+        String[] split = test.split(",");
+
+        // then
+        assertAll(
+                () -> assertNotNull(split)
+                , () -> assertEquals(split.length,1)
+        );
+    }
+
+    @Test
+    @DisplayName("\"(1,2)\" 값이 주어졌을 때 String의 substring() 메소드를 활용해 ()을 제거하고 \"1,2\"를 반환하도록 구현한다.")
+    public void 요구사항2() throws Exception {
+        // given
+        String test = "(1,2)";
+        // when
+        String s = test.replaceAll("[()]", "");
+        // then
+        assertThat(s).isEqualTo("1,2");
+    }
+
+    @Test
+    @DisplayName("\"abc\" 값이 주어졌을 때 String의 charAt() 메소드를 활용해 특정 위치의 문자를 가져오는 학습 테스트를 구현한다.")
+    public void 요구사항3() throws Exception {
+        // given
+        String test = "abc";
+        // when
+        char second = test.charAt(1);
+        // then
+        assertThat(second).isEqualTo('b');
+    }
+    @Test
+    @DisplayName("String의 charAt() 메소드를 활용해 특정 위치의 문자를 가져올 때 위치 값을 벗어나면 StringIndexOutOfBoundsException이 발생하는 부분에 대한 학습")
+    public void 요구사항3_1() throws Exception {
+        // given
+        String test = "abc";
+        // when then
+        Assertions.assertThrows(
+                StringIndexOutOfBoundsException.class, () -> test.charAt(4)
+        );
```

Comment:
`given, when, then` 구조화를 하면 테스트 코드 작성의 일관성, 구조화 되어 가독성 향상, 간결함 의 장점이 있을거라고 생각되는데요.

그렇다면, 테스트 코드를 간결하게 구현하는게 우선이 아닐까? 라는 생각이 들더라구요.  
그런의미에서 공백으로 구분 처리하면 어떨까요?  
```suggestion
        String test = "abc";

        Assertions.assertThrows(
                StringIndexOutOfBoundsException.class, () -> test.charAt(4)
        );
```

Reply:
안녕하세요!
아래는 그저 요약입니다.
리뷰 감사합니다.


요약
given when then 통해 코드를 구조화하고 주석보다는 공백을 통해 gwt의 구조를 표현하면 간결함과 가독성을 올리면 좋다.